sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala

```diff
@@ -154,6 +154,217 @@ class DataFrameWindowFunctionsSuite extends QueryTest with SharedSQLContext {
+        collect_list("value").over(Window.partitionBy($"partition").orderBy($"value".desc)
+          .rowsBetween(Window.unboundedPreceding, Window.unboundedFollowing))),
+      Seq(
+        Row("a", Array("3", "2", "2", "1")),
+        Row("b", Array("3", "2", "2", "1")),
+        Row("c", Array("3", "2", "2", "1")),
+        Row("d", Array("3", "2", "2", "1")),
+        Row("e", Array("3", "2", "2", "1")),
+        Row("f", Array("11", "10")),
+        Row("g", Array("11", "10")),
+        Row("h", Array("20")),
+        Row("i", Array())))
+  }
+
+  test("collect_set in window") {
+    val df = Seq(
+      ("a", "p1", 1),
+      ("b", "p1", 2),
+      ("c", "p1", 2),
+      ("d", "p1", 3),
+      ("e", "p1", 3),
+      ("f", "p2", 10),
+      ("g", "p2", 11),
+      ("h", "p3", 20)).toDF("key", "partition", "value")
+    checkAnswer(
+      df.select(
+        $"key",
+        collect_set("value").over(Window.partitionBy($"partition").orderBy($"value")
+          .rowsBetween(Window.unboundedPreceding, Window.unboundedFollowing))),
+      Seq(
+        Row("a", Array(1, 2, 3)),
+        Row("b", Array(1, 2, 3)),
+        Row("c", Array(1, 2, 3)),
+        Row("d", Array(1, 2, 3)),
+        Row("e", Array(1, 2, 3)),
+        Row("f", Array(10, 11)),
+        Row("g", Array(10, 11)),
+        Row("h", Array(20))))
+  }
+
+  test("skewness and kurtosis functions in window") {
+    val df = Seq(
+      ("a", "p1", 1.0),
+      ("b", "p1", 1.0),
+      ("c", "p1", 2.0),
+      ("d", "p1", 2.0),
+      ("e", "p1", 3.0),
+      ("f", "p1", 3.0),
+      ("g", "p1", 3.0),
+      ("h", "p2", 1.0),
+      ("i", "p2", 2.0),
+      ("j", "p2", 5.0)).toDF("key", "partition", "value")
+    checkAnswer(
+      df.select(
+        $"key",
+        skewness("value").over(Window.partitionBy("partition").orderBy($"key")
+          .rowsBetween(Window.unboundedPreceding, Window.unboundedFollowing)),
+        kurtosis("value").over(Window.partitionBy("partition").orderBy($"key")
+          .rowsBetween(Window.unboundedPreceding, Window.unboundedFollowing))),
+      // results are checked by scipy.stats.skew() and scipy.stats.kurtosis()
+      Seq(
+        Row("a", -0.27238010581457267, -1.506920415224914),
+        Row("b", -0.27238010581457267, -1.506920415224914),
+        Row("c", -0.27238010581457267, -1.506920415224914),
+        Row("d", -0.27238010581457267, -1.506920415224914),
+        Row("e", -0.27238010581457267, -1.506920415224914),
+        Row("f", -0.27238010581457267, -1.506920415224914),
+        Row("g", -0.27238010581457267, -1.506920415224914),
+        Row("h", 0.5280049792181881, -1.5000000000000013),
+        Row("i", 0.5280049792181881, -1.5000000000000013),
+        Row("j", 0.5280049792181881, -1.5000000000000013)))
+  }
+
+  test("aggregation function on invalid column") {
+    val df = Seq((1, "1")).toDF("key", "value")
+    val e = intercept[AnalysisException](
+      df.select(
+        $"key",
+        count("invalid").over(
```

Comment:
`over()` would be enough, as partition and orderBy is ignored anyway

Reply:
Thanks, I will remove the unnecessary parts.